src/include/mpir_request.h

```diff
@@ -424,8 +440,10 @@ static inline void MPIR_Request_free(MPIR_Request * req)
 
         int pool = (req->handle & REQUEST_POOL_MASK) >> REQUEST_POOL_SHIFT;
         MPID_THREAD_CS_ENTER(POBJ, MPIR_THREAD_POBJ_HANDLE_MUTEX);
+        MPID_THREAD_CS_ENTER(VCI, (*(MPID_Thread_mutex_t *) MPIR_Request_mem[pool].lock));
```

Comment:
This VCI critical section is removed in `request: remove the VCI handle mutex for request` and then added back here 2 commits later. Is this intentional?

Reply:
`VCI handle mutex` is global for all object handle mem, here we need per-pool request mutex.

Reply:
OK, I see it now.